src/lowering/desugar.ml

```diff
@@ -500,7 +500,7 @@ and build_actor at ts self_id es obj_typ =
     let vs = fresh_vars "v" (List.map (fun f -> f.T.typ) fields) in
     blockE
       ((match call_system_func_opt "preupgrade" es obj_typ with
-        | Some call -> [ expD (primE (I.ICPerformGC) []); expD call]
+        | Some call -> [ expD call]
```

Comment:
In the integrated PR, this might change the behaviour of the classic compiler too. I'm not sure why we inserted the GC call, but it was probably to free up space to do any copying needed in preupgrade - probably a bad idea anyway....

Reply:
Hmm, perhaps this dates from the original implementation, before we did streaming stablitization direct to stable memory, and wanted to free up space for a temporary, in-memory, candid blob. Ok to discard I think.

Reply:
Good point: This has impact on the classical support in the integrated PR.
I also think that with the streaming Candid serialization, the GC call should not be necessary.